docs/migration-v8.md

```diff
@@ -1 +1,515 @@
+        return default;
+    }
+})
+.Build();
+
+// Retry forever
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    MaxRetryAttempts = int.MaxValue,
+    Delay = TimeSpan.Zero,
+})
+.Build();
+```
+<!-- endSnippet -->
+
+- To disable waiting between retries, set the `Delay` property to `TimeSpan.Zero`.
+- To retry forever, set the `MaxRetryAttempts` property to `int.MaxValue`.
+
+### Retry and wait in v7
+
+<!-- snippet: migration-retry-wait-v7 -->
+```cs
+// Retry forever
+Policy
+    .Handle<SomeExceptionType>()
+    .WaitAndRetryForever(_ => TimeSpan.FromSeconds(1));
+
+// Wait and retry multiple times
+Policy
+  .Handle<SomeExceptionType>()
+  .WaitAndRetry(3, _ => TimeSpan.FromSeconds(1));
+
+// Wait and retry multiple times with callback
+Policy
+    .Handle<SomeExceptionType>()
+    .WaitAndRetry(3, _ => TimeSpan.FromSeconds(1), onRetry: (exception, retryCount) =>
+    {
+        // Add logic to be executed before each retry, such as logging
+    });
+
+// Wait and retry forever
+Policy
+    .Handle<SomeExceptionType>()
+    .WaitAndRetryForever(_ => TimeSpan.FromSeconds(1));
+```
+<!-- endSnippet -->
+
+### Retry and wait in v8
+
+<!-- snippet: migration-retry-wait-v8 -->
+```cs
+// Wait and retry multiple times
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    MaxRetryAttempts = 3,
+    Delay = TimeSpan.FromSeconds(1),
+    BackoffType = DelayBackoffType.Constant
+})
+.Build();
+
+// Wait and retry multiple times with callback
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    MaxRetryAttempts = 3,
+    Delay = TimeSpan.FromSeconds(1),
+    BackoffType = DelayBackoffType.Constant,
+    OnRetry = args =>
+    {
+        // Add logic to be executed before each retry, such as logging
+        return default;
+    }
+})
+.Build();
+
+// Wait and retry forever
+new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
+{
+    ShouldHandle = new PredicateBuilder().Handle<SomeExceptionType>(),
+    MaxRetryAttempts = int.MaxValue,
+    Delay = TimeSpan.FromSeconds(1),
+    BackoffType = DelayBackoffType.Constant
+})
+.Build();
+```
+<!-- endSnippet -->
+
+- It's important to remember that the configuration in v8 is options based, i.e. `RetryStrategyOptions` are used.
+- Notice that to disable waiting between retries the `Delay` property is set to `TimeSpan.Zero`.
+- To retry forever set `MaxRetryAttempts` to `int.MaxValue`.
+
+## Migrating bulkhead policy
+
+The bulkhead policy is now replaced by the [rate limiter strategy](strategies/rate-limiter.md) which uses the [`System.Threading.RateLimiting`](https://www.nuget.org/packages/System.Threading.RateLimiting) package. The new counterpart to bulkhead is `ConcurrencyLimiter`.
```

Comment:
I would probably put an extra sentence here about what is the difference between Bulkhead and Rate-limiting. Why is it that its suddenly called something different, it feels like that there must be lots of changes but there are just 2 sentences that don't say much in that regard.

Reply:
Done.